flink-runtime/src/main/java/org/apache/flink/runtime/codegeneration/SorterTemplateModel.java

```diff
@@ -0,0 +1,276 @@
+
+		// generate swap function string
+		String swapProcedures  = generateSwapProcedures();
+		String writeProcedures = generateWriteProcedures();
+		String compareProcedures = generateCompareProcedures();
+
+		templateVariables.put("writeProcedures", writeProcedures);
+		templateVariables.put("swapProcedures", swapProcedures);
+		templateVariables.put("compareProcedures", compareProcedures);
+
+
+		return templateVariables;
+	}
+
+	public boolean isSortingKeyFixedSize(){
+		System.out.println(typeComparator.getNormalizeKeyLen());
+		return true;
+	}
+
+	private ArrayList<Integer> generatedSequenceFixedByteOperators(int numberBytes){
+		ArrayList<Integer> operators = new ArrayList<>();
+		if( numberBytes > NormalizedKeySorter.DEFAULT_MAX_NORMALIZED_KEY_LEN ) {
+			return operators;
+		}
+
+		// also include offset
+		numberBytes += NormalizedKeySorter.OFFSET_LEN;
+
+		// greedy checking index
+		int i = 0;
+		while( numberBytes > 0 ) {
+			int bytes = POSSIBLE_FIXEDBYTE_OPERRATORS[i];
+			if( bytes <= numberBytes ) {
+				operators.add(bytes);
+				numberBytes -= bytes;
+			} else {
+				i++;
+			}
+		}
+		return operators;
+	}
+
+	public ArrayList<Integer> getBytesOperators() {
+		return byteOperators;
+	}
+
+	public String getSorterName (){
+		return this.sorterName;
+	}
+
+	private String generateSwapProcedures(){
+		String procedures = "";
+
+		if( this.byteOperators.size() > 0 ) {
+			String temporaryString = "";
+			String firstSegmentString = "";
+			String secondSegmentString = "";
+
+			int accOffset = 0;
+			for( int i = 0; i  < byteOperators.size(); i++ ){
+				int numberByte = byteOperators.get(i);
+				int varIndex  = i+1;
+
+				String primitiveClass = byteOperatorMapping.get(numberByte);
+				String primitiveType  = primitiveClass.toLowerCase();
+
+				String offsetString = "";
+				if( i > 0 ) {
+					accOffset += byteOperators.get(i-1);
+					offsetString = "+" + accOffset;
+				}
+
+				temporaryString     += String.format("%s temp%d = segI.get%s(iBufferOffset%s);\n",primitiveType, varIndex, primitiveClass, offsetString );
+
+				firstSegmentString  += String.format("segI.put%s(iBufferOffset%s, segJ.get%s(jBufferOffset%s));\n", primitiveClass, offsetString, primitiveClass, offsetString);
+
+				secondSegmentString += String.format("segJ.put%s(jBufferOffset%s, temp%d);\n", primitiveClass, offsetString, varIndex);
+
+			}
+
+			procedures = temporaryString
+				+ "\n" + firstSegmentString
+				+ "\n" + secondSegmentString;
+		} else {
+			procedures = "segI.swapBytes(this.swapBuffer, segJ, iBufferOffset, jBufferOffset, this.indexEntrySize);";
+		}
+
+		return procedures;
+	}
+
+	private String generateWriteProcedures(){
+		String procedures = "";
+		// skip first operator for prefix
+		if( byteOperators.size() > 1 && ByteOrder.nativeOrder() == ByteOrder.LITTLE_ENDIAN ) {
+			int offset = 0;
+			for( int i = 1; i < byteOperators.size(); i++ ){
+				int noBytes = byteOperators.get(i);
+				if( noBytes == 1 ){
+					break;
```

Comment:
OK, if you don't change it to `continue`, then please write a comment here saying that `break` is OK here because ... (so that people reading the code don't have to spend time thinking about this), and also a comment to the computation of `byteOperators` saying that it is important that the 1-byte chunk comes last (so that someone changing the logic there won't accidentally break this).